tools/clang/test/CodeGenHLSL/unroll/count_negative.hlsl

```diff
@@ -0,0 +1,28 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
```

Comment:
You could still simplify count_negative and count_zero to not have the array of resources, and then have a test specifically called count_resource_array or something such.

Reply:
I need the compilation to fail in order to check the warning message. This is just a simple way of guaranteeing it.

Reply:
You should be able to see the failure with any loop body, even an empty one. Otherwise that's a bug.

Reply:
Same for unroll(0), what causes the unroll failure is the unknown loop bound g_cond, not what is inside of the loop.

Reply:
It wasn't a bad idea to still have one test specifically for  resource arrays, but I think the changes here are great!